frontend/src/services/HTTPHelper.ts

```diff
@@ -0,0 +1,368 @@
+/* eslint-disable max-classes-per-file */
+import { injectable } from 'inversify';
+import 'reflect-metadata';
+import ArgumentError from '../errors/ArgumentError';
+import CancellationError from '../utils/CancellationError';
+import { CancellationToken } from '../utils/CancellationToken';
+import nop from '../utils/nop';
+import { HTTPHelperOptions } from './HTTPHelperOptions';
+import NetworkError from './NetworkError';
+
+interface HTTPHeaderCollection {
+  [key: string]: string | undefined;
+}
+
+type URLQueryValueType = number | string | (number | string)[];
+
+interface URLQuery {
+  [key: string]: URLQueryValueType;
+}
+
+export abstract class HTTPRequest {
+  // TODO: Why is this constructor useless? It protects the class to be consistent.
+  // eslint-disable-next-line no-useless-constructor
+  public constructor(
+    public readonly headers: HTTPHeaderCollection = {},
+    public readonly query: URLQuery = {}
+  ) {}
+}
+
+export class HTTPGETRequest extends HTTPRequest {
+  // TODO: Why is this constructor useless? It protects the class to be consistent.
+  // eslint-disable-next-line no-useless-constructor
+  public constructor(headers: HTTPHeaderCollection = {}, query: URLQuery = {}) {
+    super(headers, query);
+  }
+}
+
+export class HTTPPOSTRequest extends HTTPRequest {
+  // TODO: Why is this constructor useless? It protects the class to be consistent.
+  // eslint-disable-next-line no-useless-constructor
+  public constructor(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+    public readonly body: any,
+    headers: HTTPHeaderCollection = {},
+    query: URLQuery = {}
+  ) {
+    super(headers, query);
+  }
+}
+
+export class HTTPResponse<TResult> {
+  // TODO: Why is this constructor useless? It protects the class to be consistent.
+  // eslint-disable-next-line no-useless-constructor
+  public constructor(
+    public readonly result: TResult | null,
+    public readonly headers: HTTPHeaderCollection,
+    public readonly status: number,
+    public readonly statusText: string
+  ) {}
+}
+
+function parseResponseHeaders(headers: string): HTTPHeaderCollection {
```

Comment:
Maybe extract this function.
It provides functionality that is not limited to the `HttpHelper` that _may_ (probably not:D) be reused.

Reply:
I don't see value in this currently. If this is ever needed in another place, it is very easy to move it to a new file and export it. Currently this would bloat our directory only.

Reply:
It's just a preference that comes from my experience in other languages. I don't like bloated files, I prefer files with only one class oder function over files that are named HttpHelper.ts but where i have to search for the class. My suggestion would bloat the repository, but i think you can set it up so that only the class HttpHelper is visible outside a /HttpHelper folder.

TLDR: Do it like you want :D